Windows Quickstart

kraftp opened the pull request:
Create a `.bat` version of the script starting a local Postgres docker, allowing a Windows user to run the entire quickstart natively without WSL.

jtcoyle commented on examples/hello/start_postgres_docker.bat:
This scripts works and creates the dbos-db container and starts the db but it does not catch the the db is started and keeps waiting until the batch times out.  I have tried everything I can have the batch detect the the db has started to no avail.  I have also added another if check to catch if the container with the same name already exists.  

@echo off

setlocal

rem Check if PGPASSWORD is set
if "%PGPASSWORD%"=="" (
  echo Error: PGPASSWORD is not set.
  exit /b 1
)

rem Start Postgres in a local Docker container
docker run --rm --name=dbos-db --env=POSTGRES_PASSWORD=%PGPASSWORD% --env=PGDATA=/var/lib/postgresql/data --volume=/var/lib/postgresql/data -p 5432:5432 -d postgres:16.1

	if !errorlevel! =="" (
	echo got to :start
	echo Error: Check if Docker container already exist
	exit /b 1
	) else (
	goto :start
	)
	
:start
rem Wait for PostgreSQL to start
echo Waiting for PostgreSQL to start...
for /l %%i in (1,1,40) do (
  docker exec dbos-db psql -U postgres -c "SELECT 1;" >NUL 2>&1
  if !LOG! == "database system is ready to accept connections" (
    echo PostgreSQL started!
    goto :break
  )
  timeout /t 1 /nobreak
)
:break

kraftp replied:
Got it, thanks!  Did you test with the most recent version of the script?  The original version containing a bug like what you described that I fixed in https://github.com/dbos-inc/dbos-ts/pull/348/commits/d7f87617bd0134f974acdb22b5fc25fef84bbbba.

jtcoyle commented:
When I ran the last .bat file I received a Do exception error.  I do have a working start.postgres_docker.bat I would like to propose.

[start_postgres_docker.zip](https://github.com/dbos-inc/dbos-ts/files/14635494/start_postgres_docker.zip)

kraftp commented:
> When I ran the last .bat file I received a Do exception error. I do have a working start.postgres_docker.bat I would like to propose.
> 
> [start_postgres_docker.zip](https://github.com/dbos-inc/dbos-ts/files/14635494/start_postgres_docker.zip)

Thank you!  I've added your fixes and updated the PR.

devhawk commented:
We should update the examples/hello readme

devhawk commented:
```cmd
> npx dbos-sdk migrate     
2024-03-18 18:32:25 [info]: Starting migration: creating database undefined if it does not exist 
'createdb' is not recognized as an internal or external command,
operable program or batch file.
2024-03-18 18:32:25 [error]: Error creating database: Command failed: createdb -h localhost -p 5432 undefined -U postgres -ew undefined
'createdb' is not recognized as an internal or external command,
operable program or batch file.

    at GlobalLogger.error (C:\Users\harry\Source\dbos\sdk-ts\dist\src\telemetry\logs.js:63:65)
    at migrate (C:\Users\harry\Source\dbos\sdk-ts\dist\src\dbos-runtime\migrate.js:31:24)
    at runAndLog (C:\Users\harry\Source\dbos\sdk-ts\dist\src\dbos-runtime\cli.js:85:28)
    at Command.<anonymous> (C:\Users\harry\Source\dbos\sdk-ts\dist\src\dbos-runtime\cli.js:49:33)
    at Command.listener [as _actionHandler] (C:\Users\harry\Source\dbos\sdk-ts\node_modules\commander\lib\command.js:494:17)
    at C:\Users\harry\Source\dbos\sdk-ts\node_modules\commander\lib\command.js:1296:65
    at Command._chainOrCall (C:\Users\harry\Source\dbos\sdk-ts\node_modules\commander\lib\command.js:1193:12)
    at Command._parseCommand (C:\Users\harry\Source\dbos\sdk-ts\node_modules\commander\lib\command.js:1296:27)
    at C:\Users\harry\Source\dbos\sdk-ts\node_modules\commander\lib\command.js:1082:27
    at Command._chainOrCall (C:\Users\harry\Source\dbos\sdk-ts\node_modules\commander\lib\command.js:1193:12)
```

devhawk commented:
```cmd
npx dbos-sdk migrate     
2024-03-18 18:32:25 [info]: Starting migration: creating database undefined if it does not exist 
'createdb' is not recognized as an internal or external command,
operable program or batch file.
2024-03-18 18:32:25 [error]: Error creating database: Command failed: createdb -h localhost -p 5432 undefined -U postgres -ew undefined
'createdb' is not recognized as an internal or external command,
operable program or batch file.

    at GlobalLogger.error (C:\Users\harry\Source\dbos\sdk-ts\dist\src\telemetry\logs.js:63:65)
    at migrate (C:\Users\harry\Source\dbos\sdk-ts\dist\src\dbos-runtime\migrate.js:31:24)
    at runAndLog (C:\Users\harry\Source\dbos\sdk-ts\dist\src\dbos-runtime\cli.js:85:28)
    at Command.<anonymous> (C:\Users\harry\Source\dbos\sdk-ts\dist\src\dbos-runtime\cli.js:49:33)
    at Command.listener [as _actionHandler] (C:\Users\harry\Source\dbos\sdk-ts\node_modules\commander\lib\command.js:494:17)
    at C:\Users\harry\Source\dbos\sdk-ts\node_modules\commander\lib\command.js:1296:65
    at Command._chainOrCall (C:\Users\harry\Source\dbos\sdk-ts\node_modules\commander\lib\command.js:1193:12)
    at Command._parseCommand (C:\Users\harry\Source\dbos\sdk-ts\node_modules\commander\lib\command.js:1296:27)
    at C:\Users\harry\Source\dbos\sdk-ts\node_modules\commander\lib\command.js:1082:27
    at Command._chainOrCall (C:\Users\harry\Source\dbos\sdk-ts\node_modules\commander\lib\command.js:1193:12)
```

devhawk reviewed (approved):
Looks good, though we should probably still update the readme

jtcoyle commented:
I will look at updating the readme file
